core/domain/topic_services_test.py

```diff
@@ -67,6 +68,105 @@ def setUp(self):
         self.user_admin = user_services.UserActionsInfo(self.user_id_admin)
 
+    def test_assign_roles_with_invalid_new_role(self):
+        with self.assertRaisesRegexp(Exception, 'Invalid role: %s' % None):
+            topic_services.assign_role(
+                self.user_admin, self.user_a, None, self.TOPIC_ID)
+
+    def test_get_all_topic_rights(self):
+        topic_rights = topic_services.get_all_topic_rights()
+
+        self.assertTrue(topic_rights)
+
+    def test_apply_change_list_with_update_additional_story_ids(self):
+        story_ids = [self.story_id_1, self.story_id_2, self.story_id_3]
+        change_list = [topic_domain.TopicChange({
+            'cmd': topic_domain.CMD_UPDATE_TOPIC_PROPERTY,
+            'property_name': topic_domain.TOPIC_PROPERTY_ADDITIONAL_STORY_IDS,
+            'old_value': None,
+            'new_value': story_ids,
+        })]
+        topic = topic_services.apply_change_list(self.TOPIC_ID, change_list)[0]
+
+        self.assertTrue(topic.additional_story_ids == story_ids)
+
+    def test_apply_change_list_with_language_code_change(self):
+        language_code = 'ENG'
+        change_list = [topic_domain.TopicChange({
+            'cmd': topic_domain.CMD_UPDATE_TOPIC_PROPERTY,
+            'property_name': topic_domain.TOPIC_PROPERTY_LANGUAGE_CODE,
+            'old_value': None,
+            'new_value': language_code,
+        })]
+        topic = topic_services.apply_change_list(self.TOPIC_ID, change_list)[0]
+
+        self.assertTrue(topic.language_code == language_code)
+
+    def test_apply_change_list_with_subtopic_property_title_change(self):
+        new_title = 'new_title'
+        change_list = [topic_domain.TopicChange({
+            'cmd': topic_domain.CMD_UPDATE_SUBTOPIC_PROPERTY,
+            'property_name': topic_domain.SUBTOPIC_PROPERTY_TITLE,
+            'subtopic_id': self.subtopic_id,
+            'id': self.subtopic_id,
+            'new_value': new_title,
+            'old_value': None,
+        })]
+        topic = topic_services.apply_change_list(self.TOPIC_ID, change_list)[0]
+        subtopic_index = topic.get_subtopic_index(self.subtopic_id)
+
+        self.assertTrue(topic.subtopics[subtopic_index].title == new_title)
+
+    def test_apply_change_list_with_migrate_subtopic_schema_cmd(self):
+        change_list = [topic_domain.TopicChange({
+            'cmd': topic_domain.CMD_MIGRATE_SUBTOPIC_SCHEMA_TO_LATEST_VERSION,
+            'from_version': None,
+            'to_version': None,
+        })]
+        topic = topic_services.apply_change_list(self.TOPIC_ID, change_list)[0]
+
+        self.assertTrue(
+            topic.subtopic_schema_version ==
+            feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION)
+
+    def test_get_topic_memcache_key_with_version(self):
+        topic = topic_services.get_topic_by_id(
+            self.TOPIC_ID, version=self.topic.version)
+
+        self.assertIsNotNone(topic)
```

Comment:
Check that the retrieved topic id matches self.TOPIC_ID and the version matches self.topic.version.